pkg/gpu/cuda/env.go

```diff
@@ -0,0 +1,117 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux_bpf
+
+package cuda
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+
+	"github.com/DataDog/datadog-agent/pkg/util/kernel"
+)
+
+const cudaVisibleDevicesEnvVar = "CUDA_VISIBLE_DEVICES"
```

Comment:
Do we want it to make it configurable? This is the env variable defined by NVIDIA, I don't think it's going to change (and if it does, we'll probably have to do detection on a case-by-case basis).

Reply:
what do you mean by case-by-case? we would need to make a code change in the agent to support both options based on the nvidia driver version i suppose, but at least it would allow "quick fix" for the customers without upgrading an agent to a new version

Reply:
My assumption is that if that changes, it will be in the runtime itself, so depending on what runtime it's the program linked against we should have one or other variable being used, and we might even have different runtimes running with the same agent (e.g., workloads in different containers being compiled with different versions of the library). I'm also assuming that if the variable changes, it's not going to be just the name but the format too. In other words, my fear is that we're adding more code that will end up not solving the problem if it happens.